pmm/kubernetes-cluster-staging.groovy

```diff
@@ -0,0 +1,372 @@
+
+                    SSHLauncher ssh_connection = new SSHLauncher(env.IP, 22, 'aws-jenkins')
+                    DumbSlave node = new DumbSlave(env.VM_NAME, "spot instance job", "/home/ec2-user/", "1", Mode.EXCLUSIVE, "", ssh_connection, RetentionStrategy.INSTANCE)
+
+                    Jenkins.instance.addNode(node)
+                }
+                node(env.VM_NAME){
+                    sh """
+                        set -o errexit
+                        set -o xtrace
+
+                        if [ -n "$SSH_KEY" ]; then
+                            echo '$SSH_KEY' >> /home/ec2-user/.ssh/authorized_keys
+                        fi
+
+                        sudo yum -y update --security
+                        sudo yum -y install git svn docker
+                        sudo amazon-linux-extras install epel -y
+                        sudo usermod -aG docker ec2-user
+                        sudo systemctl start docker
+                        sudo mkdir -p /srv/pmm-qa || :
+                        pushd /srv/pmm-qa
+                            sudo git clone --single-branch --branch \${PMM_QA_GIT_BRANCH} https://github.com/percona/pmm-qa.git .
+                            sudo git checkout \${PMM_QA_GIT_COMMIT_HASH}
+                        popd
+                    """
+                }
+                script {
+                    def node = Jenkins.instance.getNode(env.VM_NAME)
+                    Jenkins.instance.removeNode(node)
+                    Jenkins.instance.addNode(node)
+                }
+                archiveArtifacts 'IP'
+            }
+        }
+        stage('Setup Minikube') {
+            steps {
+                script {
+                    withEnv(['JENKINS_NODE_COOKIE=dontKillMe']) {
+                        sh """
+                        export IP=\$(cat IP)
+                        export VM_NAME=\$(cat VM_NAME)
+                        """
+                        node(env.VM_NAME){
+                            sh """
+                                set -o errexit
+                                set -o xtrace
+                                sudo yum -y install curl
+                                sudo curl -Lo /usr/local/sbin/minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+                                sudo chmod +x /usr/local/sbin/minikube
+                                sudo ln -s /usr/local/sbin/minikube /usr/sbin/minikube
+                                alias kubectl='minikube kubectl --'
+                                sleep 5
+                            """
+                        }
+                    }
+                }
+            }
+        }
+        stage('Run Operator Setup Script') {
+            steps {
+                script {
+                    withEnv(['JENKINS_NODE_COOKIE=dontKillMe']) {
+                        sh """
+                            export IP=\$(cat IP)
```

Comment:
IP is not used

Reply:
fixed, https://github.com/Percona-Lab/jenkins-pipelines/pull/744